db/compaction/compaction_job.cc

```diff
@@ -1518,20 +1030,20 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       c_iter->ResetRecordCounts();
       RecordCompactionIOStats();
     }
-
+    // for preclude_bottommost feature the output might be bottommost or
+    // preclude_bottommost, update it here
+    if (bottommost_level_ && compact_->compaction->output_level() > 1) {
+      sub_compact->SetCurrentCompactionOutput(c_iter.get());
+    }
```

Comment:
I still worried about the extra overhead of CPU it is introduced when processing every key, when preclude level is not used at all. It is safer to guard everything between line 1035 and 1037 with a `if(c_iter->preclude_bottommost() ...`

Reply:
The new code uses pre-set `SupportsPrecludeBottommost()` for checking:
https://github.com/facebook/rocksdb/blob/9c9a2bb205f959fe3ab10f180d17b674662f383b/db/compaction/compaction.cc#L353-L355